### DISCUSSION TITLE
Add `.cheatmd` support for library generation

### DISCUSSION THREAD

**Discussion by inoas** [gleam-lang/gleam#1829] [Ideas & suggestions]
References:

- https://hexdocs.pm/ex_doc/cheatsheet.html
- https://twitter.com/josevalim/status/1582770471728484352

**Comment by lpil** [comment]
Could you expand upon this please? Thanks

**Comment by inoas** [comment]
This feature would allow library developers to create one or many cheatsheets which.

Here is a cheatsheet on how a cheatsheet would look like: https://hexdocs.pm/ex_doc/cheatsheet.html
Here is the source for the cheatsheet: https://github.com/elixir-lang/ex_doc/blob/v0.29.0/Cheatsheet.cheatmd#L1

For every cheatsheet (usually just one) the cheatsheet's `h1`/`#` defines the link title of the cheat sheet in the menu bar of the generated docs.

Cheatsheets seem to be regular markdown files with some custom css classes which we would ship the styling for as part of the docs CSS. We would use our current styling and just add some support for having multiple columns (responsive, try it on <https://hexdocs.pm/ex_doc/cheatsheet.html>).

I suggest (for now) we just use the same grid utilities that elixir's `.cheatmd` supports and just give it different colouration / fonts (if not already just inherited on the base page's CSS)

Cheatsheets generally allow a quick overview of a library.

https://hexdocs.pm/ex_doc/cheatsheet.html#with-headings-and-code-links also shows that code links would be a thing.

If this should be done I would attempt implementing it and just reimplement what the elixir folks did (in Rust instead of Elixir, HTML, CSS, JS, I suppose)

**Reply by lpil** [reply]
Thanks! is cheatmd established outside of Elixir? If it's not a common thing then I think it's worth looking at what else it out there before we commit to this. It seems like quite a niche use case so I'm not sold on the value just yet.
